### ISSUE TITLE
Wisdom integration hardcodes one local install and terminates other WeaselServer instances

### ISSUE THREAD

**Issue by Yi-frank-phy** [Yi-frank-phy/neural-weasel#16]
## Severity

Medium-high integration/safety issue.

The current Wisdom startup script is machine-specific and can terminate unrelated Weasel instances.

## Evidence

`start-neural-weasel-integration.ps1` hard-codes:

```powershell
$compatibilityRoot = 'C:\输入法\wisdom_weasel_installer\weasel-release'
$compatibilityShell = Join-Path $compatibilityRoot 'WeaselServer.exe'
```

It then enumerates every process named `WeaselServer.exe` and stops any instance whose executable path is not exactly that hard-coded Wisdom shell:

```powershell
foreach ($process in @(Get-CimInstance Win32_Process -Filter "Name='WeaselServer.exe'")) {
    if ($process.ExecutablePath -and
        [IO.Path]::GetFullPath($process.ExecutablePath) -eq $desiredPath) {
        $desiredShellRunning = $true
        continue
    }
    Stop-Process -Id $process.ProcessId -ErrorAction Stop
}
```

This conflicts with the repository's isolation goal: an official Weasel, another experimental profile, or another user's intended Weasel shell can be forcibly exited simply because it shares the executable name.

The hard-coded absolute path also means the checked-in integration is not reproducible on another installation layout without source/script editing.

## Required work

1. Remove the repository-level hard-coded `C:\输入法\...` path.
2. Resolve the intended Wisdom shell from an explicit install manifest/configuration produced by the integration installer.
3. Never terminate arbitrary `WeaselServer.exe` processes by filename alone.
4. If the Wisdom shell genuinely requires exclusive ownership, verify an exact registered profile/install identity and limit process actions to the exact managed instance.
5. Prefer connection/restart logic that leaves unrelated Weasel installations untouched.
6. Add a dry-run regression with at least two fake/distinct Weasel executable paths and prove the unmanaged instance is never selected for termination.

## Acceptance criteria

- The integration works from a non-hard-coded install root.
- Starting Neural Weasel does not kill an unrelated official/experimental Weasel process.
- Managed-process identity is derived from installation state, not executable basename alone.
- The behavior is covered by an install/integration safety test.

**Comment by Yi-frank-phy** [comment]
PR #14 now removes the immediate destructive/machine-specific behavior: `start-neural-weasel-integration.ps1` no longer hard-codes `C:\输入法\...`, accepts `-CompatibilityRoot` / `NEURAL_WEASEL_WISDOM_ROOT`, never calls `Stop-Process` on unrelated Weasel instances, and only treats the backend as ready when model/int8/full-logits/runtime-index identity matches. I am leaving #16 open because the stronger acceptance criterion—deriving the Wisdom shell identity from installer-managed configuration/manifest rather than an explicit parameter/environment variable—is not yet implemented.
